### DISCUSSION TITLE
add event to custom element

### DISCUSSION THREAD

**Discussion by patrickwasp** [zauberzeug/nicegui#4085] [Q&A] [answered]
### Question

I'd like to emit an event from a custom element. I'm not sure how to best do that. I already emit a javascript event and listen for it within the element, but I'm not sure how to propagate that to the parent of the element. 

```python
import nicegui.element
import nicegui.events


class CustomSvg(nicegui.element.Element, component="custom_svg.js"):
    def __init__(self, svg_content: str) -> None:
        super().__init__()
        self._props["svgContent"] = svg_content
        self.on("svg-clicked", self.on_svg_clicked)

    def on_svg_clicked(self, event: nicegui.events.GenericEventArguments) -> None:
        element_info = event.args
        print(f"Clicked element: {element_info}")
        # emit event here?

    def update_svg(self, svg_content: str) -> None:
        self._props["svgContent"] = svg_content
        self.update()
```

then I'd like to call it from another place something like this

```python
svg="<svg></svg>"
my_custom_svg = CustomSvg(svg).on('on_svg_clicked', lambda: ui.notify(element_info))
```

**Comment by falkoschindler** [comment, marked as answer]
Hi @patrickwasp,

Have a look into our ["Custom Vue Component" example](https://github.com/zauberzeug/nicegui/blob/main/examples/custom_vue_component/). You can emit events in JavaScript via `$emit(...)` and receive them in Python using `.on(...)`
https://github.com/zauberzeug/nicegui/blob/3cfb14afcc4d7ecdfb083316305350f00564bc33/examples/custom_vue_component/counter.js#L15
https://github.com/zauberzeug/nicegui/blob/3cfb14afcc4d7ecdfb083316305350f00564bc33/examples/custom_vue_component/counter.py#L11

**Reply by patrickwasp** [reply]
I based mine on that example, but for some reason got it in my head to capture the event within the python element part first. Thanks
